design/src/rfcs/rfc0007_timeout_config.md

```diff
@@ -0,0 +1,150 @@
+=======================================
+
+> Status: RFC
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+It is not currently possible for users of the SDK to configure a client's maximum number of retry attempts. This RFC
+establishes a method for users to set the number of retries to attempt when calling a service and would allow users to
+disable retries entirely. This RFC would introduce breaking changes to the `retry` module of the `aws-smithy-client`
+crate.
+
+Terminology
+-----------
+
+There's a lot of terminology to define so I've broken it up into two sections
+
+### General terms
+
+- **Smithy Client**: A `aws_smithy_client::Client<C, M, R>` struct that is responsible for gluing together the
+  connector, middleware, and retry policy. This is not generated and lives in the `aws-smithy-client` crate.
+- **Fluent Client**: A code-generated `Client<C, M, R>` that has methods for each service operation on it. A fluent
+  builder is generated alongside it to make construction easier.
+- **AWS Client**: A specialized Fluent Client that defaults to using a `DynConnector`, `AwsMiddleware`, and `Standard`
+  retry policy.
+- **Shared Config**: An `aws_types::Config` struct that is responsible for storing shared configuration data that is
+  used across all services. This is not generated and lives in the `aws-types` crate.
+- **Service-specific Config**: A code-generated `Config` that has methods for setting service-specific configuration.
+  Each `Config` is defined in the `config` module of its parent service. For example, the S3-specific config struct
+  is `use`able from `aws_sdk_s3::config::Config` and re-exported as `aws_sdk_s3::Config`. In this case, "service" refers to an AWS offering like S3.
+
+### HTTP stack terms
+
+- **Service**: A trait defined in the [`tower-service` crate][tower_service::Service]. The lowest level of abstraction we deal with when making HTTP requests. Services act directly on data to transform and modify that data. A Service is what eventually turns a request into a response.
+- **Layer**: Layers are a higher-order abstraction over services that is used to compose multiple services together, creating a new service from that combination. Nothing prevents us from manually wrapping services withing services, but Layers allow us to do it in a maintainable, flexible, and generic manner. Layers don't directly act on data but produce a new compose service that does. Layers can be thought of as middleware.
+- **Middleware**: a term with several meanings,
+  - Generically speaking, middlewares are similar to Services and Layers in that they modify requests and responses.
+  - In the SDK, Middlewares have the specific meaning of being the only kind of struct that contains a `DispatchService`.
+  - Examples of Middleware include [MapRequest], [AsyncMapRequest], and [ParseResponse]. **TODO** is this true or am I getting things mixed up?
```

Comment:
I would probably add `AwsMiddleware`, the combination of all middlewares required to assemble an SDK